test/functional/tests/io_class/test_io_class_occupancy.py

```diff
@@ -353,22 +353,35 @@ def test_ioclass_occupancy_sum_cache():
         cache.purge_cache()
 
     with TestRun.step("Verify stats before IO"):
-        usage_stats_sum = IoClassUsageStats(Size(0), Size(0), Size(0))
+        usage_stats_occupancy_sum = Size.zero()
+        usage_stats_clean_sum = Size.zero()
+        usage_stats_dirty_sum = Size.zero()
+
         for i in io_classes:
-            usage_stats_sum += get_io_class_usage(cache, i.id)
-        usage_stats_sum += get_io_class_usage(cache, default_ioclass_id)
+            io_class_usage_stats = get_io_class_usage(cache, i.id)
+            usage_stats_occupancy_sum += io_class_usage_stats.occupancy
+            usage_stats_clean_sum += io_class_usage_stats.clean
+            usage_stats_dirty_sum += io_class_usage_stats.dirty
+
+        io_class_usage_stats = get_io_class_usage(cache, default_ioclass_id)
+        usage_stats_occupancy_sum += io_class_usage_stats.occupancy
+        usage_stats_clean_sum += io_class_usage_stats.clean
+        usage_stats_dirty_sum += io_class_usage_stats.dirty
 
-        cache_stats = cache.get_statistics().usage_stats
-        cache_stats.free = Size(0)
+        cache_usage_stats = cache.get_statistics().usage_stats
+        cache_usage_stats.free = Size.zero()
 
         if (
-            cache_stats.occupancy != usage_stats_sum.occupancy
-            or cache_stats.clean != usage_stats_sum.clean
-            or cache_stats.dirty != usage_stats_sum.dirty
+            cache_usage_stats.occupancy != usage_stats_occupancy_sum
+            or cache_usage_stats.clean != usage_stats_clean_sum
+            or cache_usage_stats.dirty != usage_stats_dirty_sum
         ):
             TestRun.LOGGER.error(
-                "Initial cache usage stats doesn't match sum of ioclasses stats\n"
-                f"cache stats: {cache_stats}, sumed up stats {usage_stats_sum}\n"
+                "Initial cache usage stats doesn't match sum of io classes stats\n"
+                f"Cache usage stats: {cache_usage_stats}\n"
+                f"Usage stats occupancy sum: {usage_stats_occupancy_sum}\n"
+                f"Usage stats clean sum: {usage_stats_clean_sum}\n"
+                f"Usage stats dirty sum: {usage_stats_dirty_sum}\n"
                 f"particular stats {[get_io_class_usage(cache, i.id) for i in io_classes]}"
```

Comment:
collect io_class_usage in the loop in line 360, instead of reading them again (just in case anything changes between the loop and printing message)